make/init.mk

```diff
@@ -0,0 +1,56 @@
+ROOT_DIR              := $(abspath $(dir $(lastword $(MAKEFILE_LIST)))/../)
+
+GO                    := GO111MODULE=on go
+
+CACHE_BASE            ?= $(ROOT_DIR)/.cache
+CACHE                 := $(CACHE_BASE)
+CACHE_BIN             := $(CACHE)/bin
+CACHE_INCLUDE         := $(CACHE)/include
+CACHE_VERSIONS        := $(CACHE)/versions
+CACHE_NODE_MODULES    := $(CACHE)/node_modules
+CACHE_NODE_BIN        := $(CACHE_NODE_MODULES)/.bin
+
+include $(ROOT_DIR)/.makerc
+
+# setup .cache bins first in paths to have precedence over already installed same tools for system wide use
+PATH                       := "$(PATH):$(CACHE_BIN):$(CACHE_NODE_BIN)"
+
+BUF_VERSION                ?= 0.35.1
+PROTOC_VERSION             ?= 3.13.0
+PROTOC_GEN_COSMOS_VERSION  ?= v0.3.1
+PROTOC_SWAGGER_GEN_VERSION ?= v2.2.0
+GRPC_GATEWAY_VERSION       ?= 1.14.7
+GOLANGCI_LINT_VERSION      ?= v1.38.0
+GOLANG_VERSION             ?= 1.16.1
+GOLANG_CROSS_VERSION       := v$(GOLANG_VERSION)
+STATIK_VERSION             ?= v0.1.7
+GIT_CHGLOG_VERSION         ?= v0.10.0
+MODVENDOR_VERSION          ?= v0.3.0
+MOCKERY_VERSION            ?= 2.5.1
+K8S_CODE_GEN_VERSION       ?= v0.19.3
+
+# <TOOL>_VERSION_FILE points to the marker file for the installed version.
+# If <TOOL>_VERSION_FILE is changed, the binary will be re-downloaded.
+PROTOC_VERSION_FILE             = $(CACHE_VERSIONS)/protoc/$(PROTOC_VERSION)
+GRPC_GATEWAY_VERSION_FILE       = $(CACHE_VERSIONS)/protoc-gen-grpc-gateway/$(GRPC_GATEWAY_VERSION)
+PROTOC_GEN_COSMOS_VERSION_FILE  = $(CACHE_VERSIONS)/protoc-gen-cosmos/$(PROTOC_GEN_COSMOS_VERSION)
+STATIK_VERSION_FILE             = $(CACHE_VERSIONS)/statik/$(STATIK_VERSION)
+MODVENDOR_VERSION_FILE          = $(CACHE_VERSIONS)/modvendor/$(MODVENDOR_VERSION)
+GIT_CHGLOG_VERSION_FILE         = $(CACHE_VERSIONS)/git-chglog/$(GIT_CHGLOG_VERSION)
+MOCKERY_VERSION_FILE            = $(CACHE_VERSIONS)/mockery/v$(MOCKERY_VERSION)
+K8S_CODE_GEN_VERSION_FILE       = $(CACHE_VERSIONS)/k8s-codegen/$(K8S_CODE_GEN_VERSION)
+
+MODVENDOR                       = $(CACHE_BIN)/modvendor
+SWAGGER_COMBINE                 = $(CACHE_NODE_BIN)/swagger-combine
```

Comment:
This variable isn't used anywhere that I've found, installing it to this location doesn't do anything. I have to install nodejs, then use `npm -g` to install this package globally on my machine so the associated scripts run

Reply:
it used by `script/protoc-swagger-gen.sh`

Reply:
The line you're referring to must be this one 

```
swagger-combine ./client/docs/config.json
```

I can't find anything that actually causes the installed `swagger-combine` to be used, unless you made some change in this PR. It picks up off the users global path variable and that is it